ExchangeUnion/xud-docker
# chore: mainnet uses fixed version connext and arby

This PR adds two new images which will be used in mainnet.
* arby:0.1.0 (`22a3f5135263ab3532b9089de0c4ce910c77b8d3`)
* connext:7.0.0-alpha.10 (`f66093525b5f64a9dcbb1a29ebe64517bdc0698b`)

And the application metadata logic is also refined in this PR. The general idea is:
1. Tag builder stage image with original build tag suffixed with `__builder`. E.g. exchangeunion/arby:latest__connext-arby-fixed-version__x86_64__builder
2. Docker exec command `git rev-parse HEAD` to get Git revision
3. Docker exec command `git rev-parse --abbrev-ref HEAD` to get Git branch. This will break when builder stage git checkout with a particular commit. So a fallback plan will be used which parses `.git/FETCH_HEAD`.

Resolves #536 

### How to test?

1. Run `base setup.sh -b connext-arby-fixed-version`
2. Choose mainnet
3. You should see arby and connect container are using version tag images
